modules/3d/src/pointcloud/load_point_cloud.cpp

```diff
@@ -0,0 +1,199 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+
+#include "coders_base.hpp"
+#include "coders_obj.hpp"
+#include "coders_ply.hpp"
+#include "utils.hpp"
+#include "opencv2/core/utils/filesystem.private.hpp"
+#include <opencv2/core/utils/logger.hpp>
+
+#include <memory>
+
+namespace cv { namespace pc {
+
+#if OPENCV_HAVE_FILESYSTEM_SUPPORT
```

Comment:
Not sure why we have that condition here.
See also: https://github.com/opencv/opencv/issues/15675

Reply:
According to discussion w/ @asmorkalov:
> OpenCV can be built without FS support for iOS and some other bare-metal configurations. The behavior is managed by `OPENCV_HAVE_FILESYSTEM_SUPPORT` macro.
> 
> * See filesystem support in core: https://github.com/opencv/opencv/blob/master/modules/core/src/utils/filesystem.cpp
> * Please add alternative branch with "not supported" exception like here: https://github.com/opencv/opencv/blob/master/modules/core/src/glob.cpp#L286. API call should not disappear from interface.

here: https://github.com/opencv/opencv/pull/20471#discussion_r678880179